_posts/2021-02-23-OpenSSL.markdown

```diff
@@ -0,0 +1,83 @@
+---
+title: Switching back to OpenSSL
+layout: post
+---
+
+The Void Linux team is [switching back to
+OpenSSL](https://github.com/void-linux/void-packages/pull/21056) on
+**DATETOBEDETERMINED**.
+
+For most users, there should be no noticeable change. If you have any packages
+installed that are no longer provided by Void, or your system has explicit
+dependencies on LibreSSL, you will of course need to take action to ensure your
+system continues to function after the switch.
+
+---
+
+The discussion started in a Request For Comments (RFC),
+[void-linux/void-packages#20935](https://github.com/void-linux/void-packages/issues/20935),
+from April 12th, 2020, and has continued with most maintainers expressing
+support for the idea. The main reasons for the switch are as listed in the RFC:
+
+- Because most software targets OpenSSL, Void will no longer have to maintain
+  (in some cases, very complex) patches to adapt packages to LibreSSL. Such
+  patches can also be risky, due to the complexity of the OpenSSL API. The
+  results can vary between runtime errors and actual security isses.
+- Extensive support for platform specific optimizations outside of x86.
+- Access to new standards and algorithms earlier, such as full TLS 1.3 support.
+
+As a result of the above, the switch to OpenSSL is expected to lessen the time
+and effort spent on packages that require it. This is especially notable
+because most other distributions which used LibreSSL have dropped it, so there
+aren't as many people amongst whom to distribute the effort.
+
+- Alpine, which switched to LibreSSL temporarily, switched back to OpenSSL in
+  January 2019, with the [3.9.0
+  release](https://alpinelinux.org/posts/Alpine-3.9.0-released.html).
+- More recently, Gentoo, which used to offer LibreSSL as an option, has
+  [discontinued that
+  support](https://www.gentoo.org/support/news-items/2021-01-05-libressl-support-discontinued.html)
+  as well.
+
+For further context, LWN [covered the subject of LibreSSL on
+Linux](https://lwn.net/Articles/841664/) at the beginning of this year. The
+article also covers the attention and improvements that the OpenSSL project has
+received post-Heartbleed, which was one of the main reasons for Void's initial
+switch to LibreSSL.
+
+To give some examples of extra work caused by packages that expect OpenSSL, we
+need only to look at the Qt toolkit: Void's version of Qt5 is heavily patched
+in order to properly support LibreSSL. Furthermore, work on porting Qt6 hasn't
+started yet, and won't be accepted by upstream in any manner, so it would be a
+maintenance burden for as long as Qt6 receives updates.
```

Comment:
This isn't very clear. The entire Qt6 thing can probably be shortened to "The new Qt6 would need significant effort to be patched and maintained to use LibreSSL". The first sentence is wordy, can probably be shortened to "For example, Void's version of the Qt toolkit ...."

Reply:
This makes me lose the tie in into the joke about Qt receiving updates, but I can live with it :P